pkg/ansible/watches/watches.go

```diff
@@ -136,10 +136,53 @@ func (w *Watch) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	w.WatchClusterScopedResources = tmp.WatchClusterScopedResources
 	w.Finalizer = tmp.Finalizer
 	w.AnsibleVerbosity = getAnsibleVerbosity(gvk, ansibleVerbosityDefault)
+	w.addRolePlaybookPaths()
 
 	return nil
 }
 
+// addRolePlaybookPaths will add the full path for rules an playbooks
+// when the relative one is informed. Note that these files are moved in the build
+// to /opt/ansible always
+func (w *Watch) addRolePlaybookPaths() {
+	w.Playbook = getFullPath(w.Playbook)
+	w.Role = getFullPath(w.Role)
+	if w.Finalizer != nil && len(w.Finalizer.Role) > 0 {
+		w.Finalizer.Role = getFullPath(w.Finalizer.Role)
+	}
+	if w.Finalizer != nil && len(w.Finalizer.Playbook) > 0 {
+		w.Finalizer.Playbook = getFullPath(w.Finalizer.Playbook)
+	}
+}
+
+// getFullPath will return a valid full path for the role and/or playbook
+func getFullPath(path string) string {
+	if isPathRelative(path) {
+		if k8sutil.IsRunModeLocal() {
+			return addCurrentDirPath(path)
+		}
+		return addDefautImagePath(path)
+	}
+	return path
+}
+
+// isPathRelative returns true when the path informed is relative
+func isPathRelative(path string) bool {
+	return len(path) > 0 && !strings.HasPrefix(path, "/")
+}
+
+// addDefautImagePath will concat the relative path with the dir path were
+// the roles and playbooks are copied in the images
+func addDefautImagePath(path string) string {
+	return "/opt/ansible/" + path
+}
```

Comment:
We should not hardcode the `/opt/ansible` path in our watches loader. A user could theoretically change `${HOME}` in their Dockerfile for the hybrid Ansible operator such that files are written to a different directory.

See previous comment for what to do instead:
> Instead, we should always use the path relative to the current directory and then make sure that when we're running in a cluster that our current working directory contains the files we expect, based on their relative paths.

Reply:
The local relative path will not exist in the image. The watch feature is looking for the data inside of the image.